SignalRServiceBenchmarkPlugin/framework/master/Program.cs

```diff
@@ -22,58 +22,99 @@ static async Task Main(string[] args)
 
-            // Generate rpc channels
-            var channels = CreateRpcChannels(argsOption.SlaveList, argsOption.RpcPort);
-
             // Create rpc clients
-            var clients = CreateRpcClients(channels);
+            var clients = CreateRpcClients(argsOption.SlaveList, argsOption.RpcPort);
 
             // Check rpc connections
             WaitRpcConnectSuccess(clients);
+
+            // Load benchmark configuration
+            var configuration = Util.ReadFile(argsOption.BenchmarkConfiguration);
+            var benchmarkConfiguration = new BenchmarkConfiguration(configuration);
+
+            // Install plugin in master and slaves
+            await InstallPlugin(clients, benchmarkConfiguration.ModuleName);
+
+            // Process pipeline
+            await ProcessPipeline(benchmarkConfiguration.Pipeline, clients);
+        }
+        
+        private static IList<IRpcClient> CreateRpcClients(IList<string> slaveList, int rpcPort)
+        {
+            var clients = new List<IRpcClient>();
+            foreach(var slave in slaveList)
+            {
+                var client = new RpcClient().Create(slave, rpcPort);
+                clients.Add(client);
+            }
+            return clients;
         }
 
-        private static ArgsOption ParseArgs(string[] args)
+        private static async Task InstallPlugin(IList<IRpcClient> clients, string moduleName)
         {
-            Log.Information($"Parse arguments...");
-            var argsOption = new ArgsOption();
-            var result = Parser.Default.ParseArguments<ArgsOption>(args)
-                .WithParsed(options => argsOption = options)
-                .WithNotParsed(error => 
-                {
-                    Log.Information($"Parse arguments...");
-                    Environment.Exit(128);
-                });
-            return argsOption;
+            Log.Information($"Install plugin...");
+            InstallPluginInMaster(moduleName);
+            await InstallPluginInSlaves(clients, moduleName);
+        }
+
+        private static void InstallPluginInMaster(string moduleName)
+        {
+            var type = Type.GetType(moduleName);
+            _plugin = (IPlugin)Activator.CreateInstance(type);
         }
 
-        private static List<Channel> CreateRpcChannels(IList<string> slaveList, int rpcPort)
+        private static async Task InstallPluginInSlaves(IList<IRpcClient> clients, string moduleName)
         {
-            Log.Information("Open channel to rpc servers...");
-            var channels = new List<Channel>(slaveList.Count);
-            for (var i = 0; i < slaveList.Count; i++)
+            Log.Information($"Install plugin in slaves...");
+
+            var tasks = new List<Task<bool>>();
+
+            // Try to install plugin
+            foreach (var client in clients)
+            {
+                tasks.Add(client.InstallPluginAsync(moduleName));
+            }
+            await Task.WhenAll(tasks);
```

Comment:
considering: `await Task.WhenAll(from client in clients select client.InstallPluginAsync(moduleName);`

Reply:
Got it, thanks